src/System.Data.SqlClient/tests/ManualTests/SQL/CommandCancelTest/CommandCancelTest.cs

```diff
@@ -208,19 +208,28 @@ private static void ExecuteCommandCancelExpected(object state)
             Barrier threadsReady = stateTuple.Item3;
 
             string errorMessage = SystemDataResourceManager.Instance.SQL_OperationCancelled;
-            DataTestUtility.ExpectFailure<SqlException>(() =>
+            string errorMessageSevereFailure = SystemDataResourceManager.Instance.SQL_SevereError;
+
+            try
             {
-                threadsReady.SignalAndWait();
-                using (SqlDataReader r = command.ExecuteReader())
+                DataTestUtility.ExpectFailure<SqlException>(() =>
                 {
-                    do
+                    threadsReady.SignalAndWait();
+                    using (SqlDataReader r = command.ExecuteReader())
                     {
-                        while (r.Read())
+                        do
                         {
-                        }
-                    } while (r.NextResult());
-                }
-            }, errorMessage);
+                            while (r.Read())
+                            {
+                            }
+                        } while (r.NextResult());
+                    }
+                }, new string[] { errorMessage, errorMessageSevereFailure });
+            }
+            catch
+            {
+                Assert.True(false, $"An unchecked exception was thrown in {nameof(ExecuteCommandCancelExpected)}");
+            }
```

Comment:
Why don't we want the original exception to just propagate?  Doing it this way we lose details on what the unexpected exception was.

Reply:
This function is run on a thread, and when the exception propagates, it crashes the test. I could however print the exception stack trace in the Assert.

Reply:
The LoC where the thread is being created is https://github.com/saurabh500/corefx/blob/2125de82771c08d64dc63c3b7b57cc551fb8d8de/src/System.Data.SqlClient/tests/ManualTests/SQL/CommandCancelTest/CommandCancelTest.cs#L128

Reply:
Assert.True(false, ...) is also going to throw an exception.

Reply:
Ah, yes. I should change this.